File: clang/others/hanoi_tower.c
#include <stdio.h>

void tower_of_hanoi(
  int disk_number,
	char from_rod,
	char to_rod,
	char aux_rod
){
  if (disk_number == 1){
    printf("\n Move disk 1 from rod %c to rod %c", from_rod, to_rod);
    return;
  }

  tower_of_hanoi(disk_number-1, from_rod, aux_rod, to_rod);
  printf("\n Move disk %d from rod %c to rod %c", disk_number, from_rod, to_rod);
  tower_of_hanoi(disk_number-1, aux_rod, to_rod, from_rod);
}

int main(){
  int disk_number = 4;
  tower_of_hanoi(disk_number, 'A', 'C', 'B');
  printf("\n");
  return 0;
}

File: nim/others/hanoi_tower.nim
import strformat

proc towerOfHanoi(
  diskNumber: int,
  fromRod: char,
  toRod: char,
  auxRod: char
) =
  if diskNumber == 1:
    echo &"Move disk 1 from {fromRod} to rod {toRod}"
    return

  towerOfHanoi(diskNumber-1, fromRod, auxRod, toRod)
  echo &"Move disk {diskNumber} from rod {fromRod} to rod {toRod}"
  towerOfHanoi(diskNumber-1, auxRod, toRod, fromRod)

proc main() =
  let diskNumber = 4
  towerOfHanoi(diskNumber, 'A', 'C', 'B')
  echo()

when isMainModule:
  main()
